include/net/net_pkt.h

```diff
@@ -1801,10 +1801,9 @@ size_t net_pkt_remaining_data(struct net_pkt *pkt);
 int net_pkt_update_length(struct net_pkt *pkt, size_t length);
 
 /**
- * @brief Remove data from the packet at current location
+ * @brief Remove data from the beginning of the packet
  *
- * @details net_pkt's cursor should be properly initialized and,
- *          eventually, properly positioned using net_pkt_skip/read/write.
+ * @details net_pkt's cursor is reset by this function
```

Comment:
Yes, and that will let you optimize where net_pkt_pull() is called, like in ipv6_fragment.c : you'll be able to remove the net_pkt_cursor_init() which is being called right after net_pkt_pull() there.